driver/pace/driver/initialization.py

```diff
@@ -95,15 +105,15 @@ def get_driver_state(
         self,
         quantity_factory: pace.util.QuantityFactory,
         communicator: pace.util.CubedSphereCommunicator,
+        damping_coefficients: pace.util.grid.DampingCoefficients,
+        driver_grid_data: pace.util.grid.DriverGridData,
+        grid_data: pace.util.grid.GridData,
     ) -> DriverState:
+        # Ajda
+        # do I need to keep metric terms here?
```

Comment:
This is a bug, not your fault though.

You need to refactor `baroclinic_init.init_baroclinic_state` so it takes `GridData` instead of taking `MetricTerms`, and then remove the `metric_terms` here. Currently I think if you use a serialized grid, the state initialization here will completely ignore it and use a generated grid instead for initializing the model state.

Reply:
I refactored, but for now left the original stuff as comments, in case something breaks. Needs to be tested.